pool/pool.go

```diff
@@ -1584,6 +1584,13 @@ func formCacheKey(address string, signer neofscrypto.Signer) string {
 	return address + string(b)
 }
 
+func cacheKeyForSession(address string, signer neofscrypto.Signer, verb session.ObjectVerb, cnr cid.ID) string {
+	b := make([]byte, signer.Public().MaxEncodedSize())
+	signer.Public().Encode(b)
+
+	return address + string(b) + fmt.Sprintf("%d", verb) + cnr.String()
```

Comment:
do we really need it to be `string`? it is too expensive for a cache key calculation IMO

Reply:
Right now yes, it may be only string. But the internal _cache_ interface allows to pass `inrerface{}` as a key:
```func (c *Cache) Get(key interface{}) (value interface{}, ok bool) {```

We may rework this in a separate task.

Reply:
> allows to pass inrerface{}

any entity that can be compared, yes, so i would think about optimization for sure

> We may rework this in a separate task.

can you create it, please?

Reply:
BTW i seems like it is a good time to update the cache to a [generic](https://github.com/nspcc-dev/neofs-node/pull/2340) one (that PR or another issue)

Reply:
Sound ok, but what kind of key do you suggest in this case? I mean instead of `return address + string(b) + fmt.Sprintf("%d", verb) + cnr.String()`

Reply:
I think it's negligible compared to networking overhead. But if we have `fmt.Sprintf()` anyway, then maybe it should be created with it as a whole, not mixing printf and string concatenation.

Reply:
> what kind of key do you suggest in this case?

some aggregating struct i guess. but it seems that public key should be serialized anyway. well all least let it be as @roman-khimov suggested

Reply:
Updated to `fmt.Sprintf()`